server/plpgsql/statements.go

```diff
@@ -204,3 +197,29 @@ func OperationSizeForStatements(stmts []Statement) int32 {
 	}
 	return total
 }
+
+// substituteVariableReferences parses the specified |expression| and replaces
+// any token that matches a variable name in the |stack| with "$N", where N
+// indicates which variable in the returned |referenceVars| slice is used.
+func substituteVariableReferences(expression string, stack *InterpreterStack) (newExpression string, referencedVars []string, err error) {
+	scanResult, err := pg_query.Scan(expression)
+	if err != nil {
+		return "", nil, err
+	}
+
+	varMap := stack.ListVariables()
+	for _, token := range scanResult.Tokens {
+		substring := expression[token.Start:token.End]
+		if _, ok := varMap[substring]; ok {
+			// TODO: There's another bug here where the same variable could
+			//       be referenced multiple times in the expression and the
+			//       length of the slice won't be the correct index.
```

Comment:
I'm not sure that I understand this comment

Reply:
actually... on second look at this code... in the `referencedVars` list, since we don't actually de-duplicate the repeated var, the code that plugs in the var index will still work fine, so I'll pull this comment out and add a test for the case I was thinking about, too.